Compile-and-Flash-STMBoard-Guide/README.adoc

```diff
@@ -0,0 +1,52 @@
+== *Compiling and Flashing STM Board*
+
+----
+This has been tested on Linux Mint and is expected to work on other Debian based distros
+----
+
+All commands will look like `this`
+
+=== Setup
+. Go to UCSolarCarTeam repo >> Recruit Resources, and clone
+. Navigate to STM32F4Discovery-Resources and run the auto installer `sudo ./install.sh`
+. This will download STLink, gcc4mbed, and CubeMX2Makefile on your computer and add gcc4mbed to your path
+. To add CubeMX2Makefile and STLink to your path, go to terminal then go to your profile to edit the PATH environment variable. If you don't know how to use vim or nano, google it. They are quite straight forward to use. `nano ~/.profile` or `vim ~/.profile`
+. Add CubeMX2Makefile to your path `export PATH=$PATH:/opt/CubeMX2Makefile`
+. Add stlink to your path `export PATH=$PATH:/opt/stlink`
+. Save, exit, and source your profile `source ~/.profile`
+. Go to the link: http://www.st.com/en/development-tools/stm32cubemx.html
+. Scroll to the bottom of the page to the *GET SOFTWARE* section and click *Get Software*
+. Follow the instructions and download CubeMX
+. Extract files from .zip file.
+. Navigate to the directory that the files were extracted
+. Run *SetupSTM32CubeMX-VERSION.linux* `./SetupSTM32CubeMX-VERSION.linux` where  _VERSION_ is the version of CubeMx that is being installed.
+. Follow the instructions of the Installation Wizard
+
+=== Making PROJECT, Creating Makefile, AND Making
+. Open CubeMX. The default path is _/home/STM32CubeMX_
+. Select New Project
+. Click on the Board Selector Tab and scroll down till you find STM32F4DISCOVERY then click Ok
+. Go to Project >> Project Settings
+. Name your project
+. Set your project location
+. Then change the Toolchain/IDE SW4STM32. 
+----
+If you don't do this CubeMX2Makefile won't work!
+----
+. Make note of the Toolchain Folder Location
+. Click Ok and download any dependencies that are needed, then click Ok again
+. Do configurations and go to Project >> Generate Code
+. Save project
+. Create the makefile  `CubeMX2Makefile /path/to/Tool/Chain/Folder`
+. Go to the directory that the Makefile was made then make the Makefile `make`
+. Navigate to the build directory and you should see PROJECT.bin, where PROJECT is the name of your project
+
+=== Flashing
+`st-flash write /path/to/project/build/PROJECT.bin 0x8000000`
+
+----
```

Comment:
I think it formats better if this goes after line 48, otherwise this entire section looks like code.